pennylane/__init__.py

```diff
@@ -230,3 +230,6 @@ def circuit():
 def version():
     """Returns the PennyLane version number."""
     return __version__
+
+
+enable_tape()
```

Comment:
This is placed at the _bottom_ of the `__init__` file, to allow the import logic to remain in non-tape mode.

Reply:
Is that necessary so that non-tape mode can still be accessible through `qml.disable_tape()`?

Reply:
No, this is because a lot of the logic that is executed above this isn't tape-mode compatible yet 😆 

I initially added `qml.enable_tape` to the _top_ of the `__init__.py` file, but it requires a _lot_ of changes throughout the code base to get it to work. I figured it would be too hard to code-review, and might introduce bugs.